galley/pkg/runtime/processing/graph.go

```diff
@@ -0,0 +1,91 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package processing
+
+import (
+	"istio.io/istio/galley/pkg/runtime/resource"
+	"istio.io/istio/pkg/mcp/snapshot"
+)
+
+// Graph handles incoming events and facilitates creation of snapshots.
+type Graph interface {
```

Comment:
Again, naming.  This is effectively a Mixin interface. Maybe `SnapshotHandler` would be more clear.